### DISCUSSION TITLE
How to preview files like the gif in the offical README?

### DISCUSSION THREAD

**Discussion by ladyrick** [jarun/nnn#1813] [Q&A]
After many search, I was repeatedly told that "nnn" has a fancy live preview feature.
But how to preview a file like the gif in offical README? I cannot find a clue anywhere.
I work on a remote ssh environment. I don't need live preview. I just want terminal preview.
<img width="899" alt="image" src="https://github.com/jarun/nnn/assets/10379994/fc774840-4b2b-43f6-8ca7-c04d85d9cf14">

**Comment by darukutsu** [comment]
https://github.com/jarun/nnn/discussions/1637

**Reply by ladyrick** [reply]
OK, thanks for your reply. It seems that nnn itself misses an important feature and provides this feature by a hard-to-config plugin. I think terminal preview should be a builtin feature.

**Reply by darukutsu** [reply]
nnn tries to be minimal. If kitty or other terminal emulator has ability to open tabbed window why repeat? You can mount remote filesystem and than use `;t` to tab it. Or if using kitty you can use [ssh kitten](https://sw.kovidgoyal.net/kitty/kittens/ssh/#opt-kitten-ssh.forward_remote_control). Read [this](https://github.com/jarun/nnn/wiki/Live-previews)

**Reply by ladyrick** [reply]
No. That’s not called minimal. With other terminal explorer, I can do this in one command. But when it comes to nnn, I have to install an extra plugin, set a bunch of env variables, enable FIFO mode, and I even must use tmux or kitty terminal to do this. All of these efforts just get equal to other tools’ out-of-box feature. The most confusing part is, nnn’s preview works only under some specific terminals. I should say, nnn is not a terminal tool. It’s a desktop tool. This is not called minimal, even is not complicated. It’s called wired.

**Reply by darukutsu** [reply]
what terminal you using?

**Reply by ladyrick** [reply]
On the other hand, terminal preview is not a big feature. Just because nnn implement the live preview feature, nnn insists to reuse this feature to do preview in terminal. It is like to use calculus to get the area of a square. It’s also wired.

**Reply by ladyrick** [reply]
A terminal tool shouldn’t assume the user’s terminal client. By the way, I use a webshell on chrome (gotty hosted maybe)

**Comment by N-R-K** [comment]
> But how to preview a file like the gif in offical README? I cannot find a clue anywhere.

The gif is using `preview-tui` plugin. https://github.com/jarun/nnn/wiki/Live-previews

> I should say, nnn is not a terminal tool.

![image](https://github.com/jarun/nnn/assets/79544946/974ab6ba-2237-4270-881b-f22592951a64)

> This is not called minimal, even is not complicated. It’s called wired.

Feel free to use something else. There are plenty of other tui file managers around.

**Reply by kaysond** [reply]
Came across this while searching for the same thing. For the directory previews, at least, you need `.iconlookup` and `NNN_ICONLOOKUP=1` as well.
